packages/amplify-cli/src/init-steps/s0-analyzeProject.ts

```diff
@@ -241,8 +244,8 @@ const getEnvName = async (context: $TSContext): Promise<string> => {
     });
   } else if (context.exeInfo.inputParams && context.exeInfo.inputParams.yes) {
     throw new AmplifyError('ProjectInitError', {
-      message: `Invalid environment name: ${context.exeInfo.inputParams.amplify.envName}`,
-      resolution: INVALID_ENV_NAME_MSG,
+      message: `Amplify project is already initialized with environment name: ${getDefaultEnv(context)}`,
+      resolution: `To create a new environment run \`amplify add env\``,
     });
   }
```

Comment:
Couple of things:

I'm struggling to understand how do we end up in this branch if we run `amplify init -y` in already initialized project.
And therefore I'm not sure if just changing this error is safe. I.e. I'm not sure if we're not missing some other case where new error doesn't make sense. Running full e2e suite may reveal where problem could be (but it's not guaranteed to find anything). Use `yarn cloud-e2e` to trigger tests.
The caller of this code `analyzeProject` seem to set `isNewProject` flag. We could potentially use it to make sure that we throw the new error in already existing project.


Another thing is. If we want to say that `Amplify project is already initialized with environment name` we shouldn't use "default env". But rather read what env is in project files. Look for `stateManager.getCurrentEnvName` to get this value.

Apart from unit testing could you check if there's any e2e test that covers this scenario (i.e. init -y in already initialized project?). If there isn't it might be worth adding one.

Reply:
Running `yarn cloud-e2e` didn't find anything but makes sense that we might be missing another case for this error so having the error separate and earlier would be best.

Good catch on "default env", in the case when running `amplify init` with a different environment name then `amplify init -y` we would want to get the current environment. This would also work in a multi-environment app as well.

I'll look into existing e2e tests and add one if there aren't any existing test for this scenario.